crates/matrix-sdk-indexeddb/src/crypto_store.rs

```diff
@@ -756,38 +750,24 @@ impl IndexeddbCryptoStore {
         Ok(())
     }
 
-    async fn tracked_users(&self) -> Result<HashSet<OwnedUserId>, CryptoStoreError> {
-        Ok(self.tracked_users_cache.to_owned().iter().map(|u| u.clone()).collect())
-    }
-
-    async fn users_for_key_query(&self) -> Result<HashSet<OwnedUserId>, CryptoStoreError> {
-        Ok(self.users_for_key_query_cache.iter().map(|u| u.clone()).collect())
-    }
-
-    async fn update_tracked_user(&self, user: &UserId, dirty: bool) -> Result<bool> {
-        let already_added = self.tracked_users_cache.insert(user.to_owned());
-
-        if dirty {
-            self.users_for_key_query_cache.insert(user.to_owned());
-        } else {
-            self.users_for_key_query_cache.remove(user);
-        }
-
+    async fn save_tracked_users(&self, users: &[(&UserId, bool)]) -> Result<()> {
         let tx = self
             .inner
             .transaction_on_one_with_mode(KEYS::TRACKED_USERS, IdbTransactionMode::Readwrite)?;
         let os = tx.object_store(KEYS::TRACKED_USERS)?;
 
-        os.put_key_val(
-            &JsValue::from_str(user.as_str()),
-            &match dirty {
-                true => JsValue::TRUE,
-                false => JsValue::FALSE,
-            },
-        )?;
+        for (user, dirty) in users {
+            os.put_key_val(
+                &JsValue::from_str(user.as_str()),
+                &match dirty {
+                    true => JsValue::TRUE,
+                    false => JsValue::FALSE,
+                },
```

Comment:
```suggestion
                &dirty.into(),
```

Reply:
This doesn't work directly as we have a `&bool`. Will use `JsValue::from()` instead.